include/openthread/link.h

```diff
@@ -216,6 +217,34 @@ typedef struct otMacCounters
      */
     uint32_t mTxRetry;
 
+    /**
+     * Histogram of number of retries for a single direct packet until success
+     * [0 retry: packet count, 1 retries: packet count, 2 retries : packet count, ...,
+     * n retries : packet count (n == max retry limit)]
+     * The size of the array is OPENTHREAD_CONFIG_MAC_DEFAULT_MAX_FRAME_RETRIES_DIRECT.
+     */
+    uint32_t mTxDirectRetrySuccess[OPENTHREAD_CONFIG_MAC_DEFAULT_MAX_FRAME_RETRIES_DIRECT];
+
+    /**
+     * The total number of unique MAC transmission packets that meet maximal retry limit for direct packets.
+     *
+     */
+    uint32_t mTxDirectMaxRetryExpiry;
+
+    /**
```

Comment:
I suggest adding a config option to allow users to enable/disable the feature to track retry histogrram.
The reason for this is to avoid the RAM/code overhead on SED/MTD devices which may be resource-limited.

Thoughts?